zoom-pan-options.js

```diff
@@ -1,58 +1,71 @@
-  return game.settings.get("zoom-pan-options", settingName)
+  return game.settings.get('zoom-pan-options', settingName);
 }
 
 function _onWheel_Override(event) {
-  const touchpad = getSetting("touchpad-scroll")
-  if (event.deltaY === 0 && !touchpad)
-    return
+  const touchpad = getSetting('touchpad-scroll');
 
   // Prevent zooming the entire browser window
-  if (event.ctrlKey) event.preventDefault();
+  if (event.ctrlKey) {
+    event.preventDefault();
+  }
 
   // Handle wheel events for the canvas if it is ready and if it is our hover target
   let hover = document.elementFromPoint(event.clientX, event.clientY);
-  if (canvas && canvas.ready && hover && (hover.id === "board")) {
+
+  if (canvas && canvas.ready && hover && hover.id === 'board') {
     event.preventDefault();
-    let layer = canvas.activeLayer;
-    let isCtrl = event.ctrlKey || event.metaKey, isShift = event.shiftKey;
-
-    // Case 1 - rotate tokens or tiles
-    if (layer instanceof PlaceablesLayer && (isShift || (isCtrl && !touchpad))) {
-      if (touchpad && isShift && isCtrl) {
-        // Shift+Ctrl on touchpad will do what Ctrl+scroll used to do (rotate in 15° increments)
-        layer._onMouseWheel({
-          deltaY: event.deltaY, // only the sign matters, actually
-          shiftKey: false,
-          ctrlKey: event.ctrlKey,
-          metaKey: event.metaKey,
-        })
-      } else {
-        layer._onMouseWheel(event)
-      }
+    const layer = canvas.activeLayer;
+    const isOsx = window.navigator.platform === 'MacIntel';
+
+    /* These can become options easily */
+    const rotateKey = event.altKey;
+    const directionModifierKey = event.shiftKey; // changes pan direction
+    const rotationModifierKey = event.shiftKey; // changes rotation amount
+    const zoomKey = event.ctrlKey || event.metaKey;
+
+    // Case 1 - handle rotation of objects
+    // default to Alt for rotation as we are moving Shift to horizontal scroll and Ctrl is reserved for pinch zooming (OSX)
+    if (layer instanceof PlaceablesLayer && (rotateKey || (rotationModifierKey && !touchpad))) {
+      // return so we don't proceed
+      return layer._onMouseWheel({
+        ctrlKey: rotateKey, // shim alt where foundry expects ctrl for rotation
+        deltaY: event.wheelDelta, // only the sign matters, use wheelDelta instead of relying on deltaY
+        metaKey: event.metaKey,
+        shiftKey: modifierKey,
+      });
     }
+
     // Case 2 - zoom the canvas (touchpad pinch, or normal scroll)
-    else if (isCtrl || !touchpad) zoom(event)
-    // Case 3 - pan the canvas (touchpad scroll)
-    else panWithTouchpad(event);
+    if (zoomKey || !touchpad) {
+      return zoom(event);
+    }
+
+    // Cast 3 - pan the canvas but flip X and Y (touchpad scroll + SHIFT and not OSX)
+    if (touchpad && !isOsx && directionModifierKey) {
+      return panWithTouchpad({
+        deltaX: event.deltaY,
+        deltaY: event.deltaX,
+      });
+    }
```

Comment:
OSX uses deltaY and deltaX instead of deltaX and deltaY? that didn't seem to happen on my macbook. is there another reason for that code change?

I'd prefer this to just be a config setting, unless it definitely only fixes problems.

Reply:
In my testing of a mouse wheel on OSX, holding Shift and scrolling populated DeltaX instead of DeltaY. You can see some details of my findings [here](https://github.com/ElfFriend-DnD/foundryvtt-canvasScroll/issues/2).

This particular block is used only by windows machines to enable Shift + Scroll to pan horizontally.